src/android/IntercomIntentService.java

```diff
@@ -0,0 +1,51 @@
+package io.intercom.android.sdk;
+
+import android.app.IntentService;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.ResolveInfo;
+import android.os.Bundle;
+
+import com.google.firebase.messaging.RemoteMessage;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+import io.intercom.android.sdk.push.IntercomPushClient;
+
+public class IntercomIntentService extends IntentService {
+    private final IntercomPushClient intercomPushClient = new IntercomPushClient();
+
+    public IntercomIntentService() {
+        super("Intercom Cordova Intent Service");
+    }
+
+    @Override
+    protected void onHandleIntent(Intent intent) {
+        if (intercomPushClient.isIntercomPush(intent.getExtras())) {
+            intercomPushClient.handlePush(getApplication(), intent.getExtras());
+            return;
+        }
+
+        Intent passThroughIntent = new Intent(intent);
+        passThroughIntent.setComponent(null);
+
+        List<ResolveInfo> services = getPackageManager().queryIntentServices(passThroughIntent, 0);
```

Comment:
This is an `@IntDef` flag, so 0 means "no flags". (It is 0 because `0 | x == x`).